src/push.ts

```diff
@@ -0,0 +1,149 @@
+  private webSocket?: ws;
+  private webSocketSubjects: { [_ in IEntityType]: Subject<{ [accountId: string]: string }> };
+
+  private started?: Promise<void>;
+  private enabledDataTypes: { [_ in IEntityType]?: boolean };
+
+  constructor({
+    client,
+    transport,
+    httpHeaders,
+  }: {
+    client: Client;
+    transport: Transport;
+    httpHeaders: { [headerName: string]: string };
+  }) {
+    this.client = client;
+    this.transport = transport;
+    this.httpHeaders = httpHeaders;
+    this.webSocketSubjects = {
+      Mailbox: new Subject(),
+      Email: new Subject(),
+      EmailSubmission: new Subject(),
+    };
+    this.enabledDataTypes = {};
+  }
+
+  public start(): Promise<void> {
+    if (!this.started) {
+      this.started = new Promise(resolve => {
+        this.transport
+          .post<{
+            value: string;
+          }>(`${this.client.getApiUrl()}/ws/ticket`, '', this.httpHeaders)
+          .then(response => {
+            const ticket = response.value;
+            const webSocket = new ws(`${this.client.getPushUrl()}?ticket=${ticket}`);
+            webSocket.onopen = () => {
+              this.webSocket = webSocket;
+              this.sendSubscriptions(webSocket);
+              resolve();
+            };
+            webSocket.onclose = () => {
+              for (const subject of Object.values(this.webSocketSubjects)) {
+                subject.complete();
+              }
+
+              delete this.webSocket;
+            };
+            webSocket.onmessage = message => {
+              const data = JSON.parse(message.data as string);
+              if (data['@type'] == 'StateChange') {
+                for (const entityType of ENTITY_TYPES) {
+                  if (this.stateChangeContainsEntityType(data, entityType)) {
+                    this.webSocketSubjects[entityType].next(
+                      this.transformStateChange(data, entityType),
+                    );
+                  }
+                }
+              }
+            };
+            webSocket.onerror = event => {
+              console.log(`Error with websocket: ${JSON.stringify(event)}`);
+              for (const subject of Object.values(this.webSocketSubjects)) {
+                subject.error(event);
+              }
+
+              delete this.webSocket;
+            };
+          });
+      });
+    }
+
+    return this.started;
+  }
+
+  public stop(): void {
+    this.webSocket?.close();
+  }
+
+  public mailbox(): Observable<{ [accountId: string]: string }> {
+    return this.pushForEntityType('Mailbox');
+  }
+
+  public email(): Observable<{ [accountId: string]: string }> {
+    return this.pushForEntityType('Email');
+  }
+
+  public emailSubmission(): Observable<{ [accountId: string]: string }> {
+    return this.pushForEntityType('EmailSubmission');
+  }
+
+  private pushForEntityType(entityType: IEntityType): Observable<{ [accountId: string]: string }> {
```

Comment:
Should allow passing an array

Reply:
Since this method is private, you do not get to use it anyway.
If needed by the calling methods, the method will be modified later.